### PR TITLE
Fixing the assignation and de-assignation of reviewers of a PR

### PR DESCRIPTION
**Pull request by agomezmoron**
Assignation with PRs doesn't work. The same problem when removing assignees (for PRs also assignees = requested reviewers).

This fixes both issues and, depending on if it's a pull_request or issue, it use one or another flow.

### CODE REVIEW

**Comment by agomezmoron** [comment]
Contribution done. The test fails internally when calling the octokit.rest.pulls.removeRequestedReviewers method but, according to [how the endpoint it works](https://octokit.github.io/rest.js/v18), the source code of the plugin-rest-endpoint-methods library and its [documentation](https://github.com/octokit/plugin-rest-endpoint-methods.js/blob/61bbc9ae4df28bf5c389f6536855c5ed2dbfe8b3/docs/pulls/removeRequestedReviewers.md), the "reviewers" attribute is needed:

Output of jest:

```
- Expected
    + Received

      Object {
        "owner": "mockOrg",
        "pull_number": 667,
        "repo": "mockRepo",
    +   "reviewers": Array [
    +     "userA",
    +     "userB",
    +   ],
      },
```

I left the test failing because I'm not a jest expert but, honestly, I'm not sure why it fails.

**Comment by pozil** [comment]
@agomezmoron thanks for the contrib.
CI complains that `package.json` and `package-lock.json` are not in sync. Can you fix this?

**Comment by agomezmoron** [comment]
Fixed @pozil . I would appreciate a check on your side too to see if I was missing something with Jest (I'm not an expert using it and I did my best).  Until that test is fixed (or commented if it's a false positive) the CI process will fail I think.

**Comment by pozil** [comment]
Sure, I'm quite busy today and tomorrow but I'll check it out on Friday.
Some of the new tests are failing and more problematic: it looks like there's a regression on the `works with project card events` test (it was passing before and the feature was functional).

**Comment by agomezmoron** [comment]
The regression has been fixed (in the card, the issue number comes always as an issue independent on the case if it's a real issue or PR). So I had to write a method to check if it's an issue (trying to get it) and, if an exception is raised internally, leave it as non issue (so it's a PR). I'm not very happy with that implementation but after checking all the REST API, it's the only way: trying to retrieve an issue (or a PR) by an ID and, if it's not exist, so is the opposite thing.

**Comment by agomezmoron** [comment]
I worked on the tests and now all of them are in green :-)

**Comment by agomezmoron** [comment]
Hi @pozil ,

Did you have the chance to have a look on it? If I can help you with anything else just let me know :).

Cheers!

**Comment by pozil** [comment]
@agomezmoron I didn't get the time to dive into it last week but I'll do it today. Stay tuned for feedback.

File: src/action.js [hunk 1]
```diff
@@ -158,19 +212,38 @@ const runAction = async (octokit, context, parameters) => {
...
+            )}`
+        );
+        await octokit.rest.issues.addAssignees({
+            owner,
+            repo,
+            issue_number: issueNumber,
+            assignees
+        });
+    } else {
+        // Assign issue
```
Comment (pozil) [inline src/action.js]:
```suggestion
        // Assign PR
```

File: .gitignore [hunk 2]
```diff
@@ -14,4 +14,6 @@ $RECYCLE.BIN/
 /.vscode
 
 # Code coverage reports
-/coverage
\ No newline at end of file
+/coverage
+
+node_modules
```
Comment (pozil) [inline .gitignore]: We need `node_modules` in git for the action to run.

File: src/action.js [hunk 3]
```diff
@@ -78,8 +125,10 @@ const runAction = async (octokit, context, parameters) => {
 
     // Get issue info from context
     let issueNumber = context.issue?.number || context.pull_request?.number;
+    let isIssue = context.issue ? true : false;
```
Comment (pozil) [inline src/action.js:127]:
```suggestion
    let isIssue = context.hasOwnProperty('issue');
```

File: src/action.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+const { getOctokit } = require('@actions/github');
```
Comment (pozil) [inline src/action.js]: Not used, please remove.
BTW this causes a linting error that fails the CI workflow so should try to run `npm run lint` locally to detect this in the future.

**Review by pozil — CHANGES_REQUESTED** [review]
Done reviewing. Thanks for the contrib once more. I've requested some minor changes.

File: .gitignore (hunk 2 above)
Comment (agomezmoron) [reply .gitignore]: Ok, I didn't know it

File: src/action.js (hunk 3 above)
Comment (agomezmoron) [reply src/action.js:127]: linter doesn't allow it ( 127:27  error  Do not access Object.prototype method 'hasOwnProperty' from target object  no-prototype-builtins)

**Comment by agomezmoron** [comment]
I applied all the changes except one that is not allowed by the linter

**Review by pozil — APPROVED** [review]
Looking good thanks.

**Pull request merged**